Replace perf_client usage with unit tests

rmccorm4 opened the pull request:
Use infer_util unit tests in place of perf_analyzer wherever possible to reduce PA dependency

CoderHam commented on qa/L0_storage_azure/test.sh:
```diff
@@ -179,20 +185,16 @@ if [ "$SERVER_PID" == "0" ]; then
 set +e
-for BACKEND in $BACKENDS; do
-    code=`curl -s -w %{http_code} -X POST localhost:8000/v2/repository/models/${BACKEND}_float32_float32_float32/load`
+for model in `ls models/`; do
```
Why is this change needed?

rmccorm4 replied:
I'm now loading everything in the model dir, not just the float32 models, in order to run the unit tests which also test the `*_object` versions of the models.

CoderHam commented on qa/L0_storage_azure/test.sh:
```diff
@@ -202,46 +204,59 @@ wait $SERVER_PID
-    cat $SERVER_LOG
-    exit 1
-fi
+# FIXME: DLIS-3893 - Currently using autocomplete will ignore the label_filename
```
We need these tests. If we cannot work around this limitation yet, I'd recommend holding off on adding these changes. You do not need label file name for non-classification output for example.  

Since these tests are our only way to validate cloud storage support, disabling them temporarily would cause testing gaps.

CoderHam commented on qa/L0_storage_azure/infer_test.py:
```diff
@@ -78,38 +79,52 @@ def _infer_exact_helper(tester,
                                correlation_id=correlation_id)
 
+        # Allow caller to setup subset of backends to test
+        test_backends = os.environ.get("INFER_TEST_BACKENDS", "").split()
```
Set this above under as TEST_BACKENDS.

CoderHam commented on qa/L0_storage_azure/test.sh:
```diff
@@ -83,18 +83,38 @@ rm -f $SERVER_LOG_BASE* $CLIENT_LOG_BASE*
 RET=0
 
 BACKENDS="graphdef savedmodel onnx libtorch plan"
```
BACKENDS=${BACKENDS:="graphdef savedmodel onnx libtorch plan"}

CoderHam commented on qa/L0_storage_azure/infer_test.py:
```diff
@@ -37,6 +37,7 @@
 
 np_dtype_string = np.dtype(object)
 
+DEFAULT_TEST_BACKENDS = ["graphdef", "savedmodel", "plan", "onnx", "libtorch"]
```
BACKENDS = os.environ.get(
    'BACKENDS',
    "graphdef savedmodel onnx libtorch plan")

rmccorm4 commented:
I think it makes sense to wait for autocomplete changes to be made first rather than adding throwaway work or temporarily disabling part of the test. Converting to draft for now.

rmccorm4 commented:
Autocomplete changes were merged and these pass in CI now.

rmccorm4 commented:
- Refactored `L0_s3_local` to use `infer_test.py`
- Moved duplicate `L0_storage_*/infer_test.py` into `qa/common/infer_test.py` where applicable (used in gitlab tests too)

rmccorm4 commented:
Passing CI :heavy_check_mark:

tanmayv25 commented on qa/L0_s3_local/test.sh:
```diff
@@ -41,17 +41,63 @@ fi
-    # Remove version policy from config.pbtxt
-    sed -i '/^version_policy/d' models/${BACKEND}_float32_float32_float32/config.pbtxt
-done
+function run_unit_tests {
```
I thing the keyword function has been deprecated in favor of fname(). However, in rest of the project we write the bash script like: `function fname() {}`. 
See here: https://github.com/triton-inference-server/server/blob/main/qa/common/util.sh#L34

Let's follow the same.

rmccorm4 commented:
Closing and opening new PR to WAR "request changes" block.
